src/pubtools/_marketplacesvm/cloud_providers/ms_azure.py

```diff
@@ -21,7 +21,9 @@
 class AzureCredentials(CloudCredentials):
     """Represent the credentials for AzureProvider."""
 
-    azure_publisher_name: str = field(alias="AZURE_PUBLISHER_NAME", validator=instance_of(str))
+    azure_publisher_name: Optional[str] = field(
+        alias="AZURE_PUBLISHER_NAME", validator=optional(instance_of(str))
+    )
```

Comment:
Could you remind me why do we make it optional instead of removing it ?
I though we do not care much about extra fields in JSON :-)

Reply:
Yeah.. Ideally we should remove but the current implementation doesn't work properly with extra fields, so I thought it would be simpler to just mark it as optional.

Another way around is to change the creds implementation to ignore extra fields, I can do that, just need to adjust things in the code then we'll be safe to remove this

Reply:
@lslebodn Fixed.

PTAL again

Reply:
**NOTE:** The `ValueError: The key/value for "AZURE_PUBLISHER_NAME" must be set.` errors are due to the old version of `cloudpub`.

To get rid of this we need to release a new version of `cloudpub` and bump here. Should I do it? It has to be done before this PR gets merged anyway (or the tooling released).

Reply:
I forgot to check the cloudpub code :-)

lets create minor cloudpub release
and bump dependency here.

that way we can remove AZURE_PUBLISHER_NAME even in the pubtool

Reply:
Working on it

Reply:
Here: https://github.com/release-engineering/cloudpub/pull/20

@jajreidy FYI

Reply:
Checks passed so I'm marking this as resolved.

@lslebodn FYI